### PR TITLE
Chapter 9: Advanced login

### PR DESCRIPTION
**Pull request by mrminht2000**
Screenshot from 2021-07-20 15-48-35
Nội dung tìm hiểu chương 9: https://docs.google.com/document/d/1it0MQJ5VXkd_E1NTl4F0dzSY9wkVkDM9OMB90BBBS0Q/edit?usp=sharing

### CODE REVIEW

**Comment by mrminht2000** [comment]
ready

File: app/models/user.rb [hunk 1]
```diff
@@ -17,6 +19,37 @@ class User < ApplicationRecord
 
   has_secure_password
 
+  class << self
+    def digest string
+      cost = if ActiveModel::SecurePassword.min_cost
+               BCrypt::Engine::MIN_COST
+             else
+               BCrypt::Engine.cost
+             end
```
Comment (ghost) [inline app/models/user.rb]:
```suggestion
      min_cost = ActiveModel::SecurePassword.min_cost
      cost = min_cost ? BCrypt::Engine::MIN_COST : BCrypt::Engine.cost
```

**Review by ghost — COMMENTED** [review]
cmt

**Comment by mrminht2000** [comment]
ready

**Review by ghost — COMMENTED** [review]
merged

**Pull request merged**